### PR TITLE
[MSHARED-1128] Update documentation with new execute method

### PR DESCRIPTION
**Pull request by slawekjaranowski**

### CODE REVIEW

File: src/site/markdown/getting-started.md [hunk 1]
```diff
@@ -129,17 +129,24 @@ For the Context Class Loader case this would mean the following dependencies are
 
 ## Run
 
-Calling `executeGoals` runs Maven with the given goals or phases and optionally some additional environment variables. It throws a `VerificationException` in case the execution is not successful (e.g. binary not found or exit code > 0). It is either using a forked JVM or is executed in the same JVM depending on the configuration.
+Calling `execute` runs Maven with the given Verifier configuration like, cli arguments, environment variables. 
+We must provide at lease one cli argument for executing.
```
Comment (kwin) [inline src/site/markdown/getting-started.md:133]:
```suggestion
At least one CLI argument must be given which specifies either the Maven phase(s) and/or the goal(s) to execute.
```

File: src/site/markdown/getting-started.md [hunk 2]
```diff
@@ -129,17 +129,24 @@ For the Context Class Loader case this would mean the following dependencies are
...
+
+It throws a `VerificationException` in case the execution is not successful (e.g. binary not found or exit code > 0). 
+It is either using a forked JVM or is executed in the same JVM depending on the configuration.
```
Comment (kwin) [inline src/site/markdown/getting-started.md:136]:
```suggestion
The method is using either a forked JVM or the same JVM for executing Maven depending on the configuration.
```

File: src/site/markdown/getting-started.md [hunk 3]
```diff
@@ -129,17 +129,24 @@ For the Context Class Loader case this would mean the following dependencies are
 
 ## Run
 
-Calling `executeGoals` runs Maven with the given goals or phases and optionally some additional environment variables. It throws a `VerificationException` in case the execution is not successful (e.g. binary not found or exit code > 0). It is either using a forked JVM or is executed in the same JVM depending on the configuration.
+Calling `execute` runs Maven with the given Verifier configuration like, cli arguments, environment variables.
```
Comment (kwin) [inline src/site/markdown/getting-started.md:132]:
```suggestion
Calling `execute` runs Maven with the given Verifier configuration like CLI arguments and environment variables. 
```

File: src/site/markdown/getting-started.md [hunk 4]
```diff
@@ -129,17 +129,24 @@ For the Context Class Loader case this would mean the following dependencies are
...
 ```
-verifier.executeGoals( "package" );
+verifier.addCliArgument( "package" )
+verifier.execute();
 ```
 
 ## Verify
 
-After executing the Maven goals there are several methods starting with prefix `verify` which allow you to check for the build result, check the log for certain contents and the existence of generated artifacts.
+After executing the Maven process there are several methods starting with prefix `verify`
```
Comment (kwin) [inline src/site/markdown/getting-started.md:145]:
```suggestion
After calling `execute` one should call one or multiple of the methods starting with prefix `verify` to
```

File: src/site/markdown/getting-started.md [hunk 5]
```diff
@@ -129,17 +129,24 @@ For the Context Class Loader case this would mean the following dependencies are
...
-verifier.executeGoals( "package" );
+verifier.addCliArgument( "package" )
+verifier.execute();
 ```
 
 ## Verify
 
-After executing the Maven goals there are several methods starting with prefix `verify` which allow you to check for the build result, check the log for certain contents and the existence of generated artifacts.
+After executing the Maven process there are several methods starting with prefix `verify` 
+which allow you to check for the build result, check the log for certain contents and the existence of generated artifacts.
```
Comment (kwin) [inline src/site/markdown/getting-started.md:146]:
```suggestion
* check for the build result
* check the log for certain contents or 
* check for the existence of generated artifacts.
```

File: src/site/markdown/getting-started.md [hunk 6]
```diff
@@ -129,17 +129,24 @@ For the Context Class Loader case this would mean the following dependencies are
...
 
-The main method `verify(boolean)` takes into consideration a file named `expected-results.txt` being located in the base directory. Each line consists of a file path (optionally prefixed by `!`) and it is automatically verified that the file exists or is missing (in case the path starts with `!`).
+The main method `verify(boolean)` takes into consideration a file named `expected-results.txt` being located in the tested project base directory.
```
Comment (kwin) [inline src/site/markdown/getting-started.md:148]:
```suggestion
The main method `verify(boolean)` takes into consideration a file named `expected-results.txt` being located in the tested project's base directory. 
```

**Review by kwin — CHANGES_REQUESTED** [review]

File: src/site/markdown/getting-started.md (hunk 1 above)
Comment (michael-o) [reply src/site/markdown/getting-started.md:133]: Do we really need that? What if `/project/build/defaultGoal` is set?

File: src/site/markdown/getting-started.md (hunk 3 above)
Comment (michael-o) [reply src/site/markdown/getting-started.md:132]: The uppercase CLI should be everywhere in the document

Comment (michael-o) [reply src/site/markdown/getting-started.md:132]: configuration, like ...

Comment (michael-o) [reply src/site/markdown/getting-started.md:132]: CLI

File: src/site/markdown/getting-started.md (hunk 1 above)
Comment (michael-o) [reply src/site/markdown/getting-started.md:133]: must be provided

**Review by michael-o — CHANGES_REQUESTED** [review]
I also agree with the suggested changes from @kwin

File: src/site/markdown/getting-started.md (hunk 1 above)
Comment (slawekjaranowski) [reply src/site/markdown/getting-started.md:133]: Maybe:

``
At least one CLI argument should be provided which specifies either the Maven phase(s) and/or the goal(s) to execute unless project has default goal.
``

Comment (michael-o) [reply src/site/markdown/getting-started.md:133]: ...has a default goal.

Sounds good to me.

**Comment by slawekjaranowski** [comment]
suggestions applied, please next round of review

File: src/site/markdown/getting-started.md [hunk 7]
```diff
@@ -129,17 +129,27 @@ For the Context Class Loader case this would mean the following dependencies are
 
 ## Run
 
-Calling `executeGoals` runs Maven with the given goals or phases and optionally some additional environment variables. It throws a `VerificationException` in case the execution is not successful (e.g. binary not found or exit code > 0). It is either using a forked JVM or is executed in the same JVM depending on the configuration.
+Calling `execute` runs Maven with the given Verifier configuration, like CLI arguments and environment variables.
```
Comment (michael-o) [inline src/site/markdown/getting-started.md:132]: `execute()`

**Review by michael-o — APPROVED** [review]
One nit, but the rest looks good.

**Review by michael-o — APPROVED** [review]
Nice improvement

File: src/main/java/org/apache/maven/shared/verifier/Verifier.java [hunk 8]
```diff
@@ -1154,8 +1156,6 @@ public void executeGoal( String goal )
      *   verifier.setEnvironmentVariable( "key2", "value2" );
      *   verifier.execute();
      * </pre>
-     *
-     * @deprecated will be removed without replacement.
      */
     public void executeGoal( String goal, Map<String, String> envVars )
```
Comment (gnodet) [inline src/main/java/org/apache/maven/shared/verifier/Verifier.java:1160]: I think it's missing a `@Deprecated` annotation.

File: src/main/java/org/apache/maven/shared/verifier/Verifier.java [hunk 9]
```diff
@@ -1165,14 +1165,14 @@ public void executeGoal( String goal, Map<String, String> envVars )
...
      * <p>
      * For replacement please use:
      * <pre>
      *   verifier.addCliArguments( "goal1", "goal2" );
      *   verifier.execute();
      * </pre>
-     *
-     * @deprecated will be removed without replacement.
      */
     public void executeGoals( List<String> goals )
```
Comment (gnodet) [inline src/main/java/org/apache/maven/shared/verifier/Verifier.java:1177]: Missing `@Deprecated` annotation.

File: src/main/java/org/apache/maven/shared/verifier/Verifier.java [hunk 10]
```diff
@@ -1205,8 +1207,6 @@ public String getExecutable()
      *   verifier.setEnvironmentVariable( "key2", "value2" );
      *   verifier.execute();
      * </pre>
-     *
-     * @deprecated will be removed without replacement.
      */
     public void executeGoals( List<String> goals, Map<String, String> envVars )
```
Comment (gnodet) [inline src/main/java/org/apache/maven/shared/verifier/Verifier.java:1211]: Missing `@Deprecated` annotation.

**Review by gnodet — CHANGES_REQUESTED** [review]

**Review by gnodet — CHANGES_REQUESTED** [review]
Btw, I've seen a few other deprecated methods which are not flagged as `@Deprecated` with a proper annotation (just the `@deprecated` javadoc tag):
 * `resetStreams()`
 * `displayStreamBuffers()`
 * `setDebug()`
 * `setMavenDebug()`
 It would be nice to add the annotation on those too !

**Comment by slawekjaranowski** [comment]
@gnodet thanks - you are right - I will add all missing annotation in separate PR.

**Comment by jira-importer** [comment]
Resolve #155

**Comment by jira-importer** [comment]
Resolve #155

**Pull request merged**